src/domain.cc

```diff
@@ -458,8 +458,9 @@ NLV_WORKER_EXECUTE(Domain, GetId)
 {
   NLV_WORKER_ASSERT_DOMAIN();
 
-  int result = virDomainGetID(Handle().ToDomain());
-  if (result == -1) {
+  unsigned int result = virDomainGetID(Handle().ToDomain());
+  if (result == -1u) {
```

Comment:
not impossible :)
virDomainGetID returns an unsigned int (and the api doc said it can be minus one), not an int. -1u is the representation of -1 as unsigned int (aka 4294967295) .. and this is what virDomainGetID returns.